engine/flink/avro-util/src/test/scala/pl/touk/nussknacker/engine/avro/schema/GeneratedAvroClassWithLogicalTypesSchema.scala

```diff
@@ -0,0 +1,62 @@
+package pl.touk.nussknacker.engine.avro.schema
+
+import io.circe.{Encoder, Json}
+import org.apache.avro.{LogicalTypes, Schema}
+import org.apache.avro.data.TimeConversions
+import org.apache.avro.specific.SpecificRecordBase
+import pl.touk.nussknacker.engine.util.json.BestEffortJsonEncoder
+
+import java.time._
+
+// To avoid name collision with GeneratedAvroClassWithLogicalTypes
+object GeneratedAvroClassWithLogicalTypesSchema extends TestSchemaWithSpecificRecord {
+
+  val date: LocalDateTime = LocalDateTime.of(2020, 1, 2, 3, 14, 15)
+
+  val instant: Instant = date.toInstant(ZoneOffset.UTC)
+
+  val decimal: java.math.BigDecimal = null // TODO: fixedEncoder does not handle avro decimalConversion
+
+  override lazy val schema: Schema = GeneratedAvroClassWithLogicalTypes.SCHEMA$
+
+  override def specificRecord: SpecificRecordBase = new GeneratedAvroClassWithLogicalTypes(
+    exampleData("text").asInstanceOf[CharSequence],
+    exampleData("dateTime").asInstanceOf[Instant],
+    exampleData("date").asInstanceOf[LocalDate],
+    exampleData("time").asInstanceOf[LocalTime],
+    exampleData("decimal").asInstanceOf[java.math.BigDecimal]
+  )
+
+  override def exampleData: Map[String, Any] = Map(
+    "text" -> "lorem ipsum",
+    "dateTime" -> instant,
+    "date" -> date.toLocalDate,
+    "time" -> date.toLocalTime,
+    "decimal" -> decimal
+  )
+
+  override def stringSchema: String = "" // see hardcoded definition in GeneratedAvroClassWithLogicalTypes
+
+  val fixedEncoder: Encoder[GeneratedAvroClassWithLogicalTypes] = new Encoder[GeneratedAvroClassWithLogicalTypes] {
```

Comment:
Why this is needed? Avro encoder doesn't support logical types? Can't it be replaced by list of encoders : date, timestamp and so on instead of fixed encoder for surrounding record?

Reply:
Avro does support logical types, but the problem is in the output json structure. ConfluentJsonPayloadDeserializer uses "allegro.schema.json2avro.converter" which requires flattened json structure of avro-converted types. So here I need to provide serializer that somehow corresponds to that mechanism.

Reply:
In `ConfluentJsonPayloadDeserializer` I've added notice to use jsonPaylod with schemas without logical types (due to avro conversion limitations). In tests I've introduced `GeneratedAvroClassSample` where schema contains some basic types, enum, records and arrays.